samples/samples/admin/backup_snippet.py

```diff
@@ -0,0 +1,556 @@
+    )
+    request = backup_pb.ListBackupsRequest(
+        parent=instance.name,
+        filter='expire_time < "{}-{}-{}T{}:{}:{}Z"'.format(*expire_time.timetuple()),
+    )
+    operations = spanner_client.database_admin_api.list_backups(request)
+    for backup in operations:
+        print(backup.name)
+
+    # List all backups with a size greater than some bytes.
+    print("All backups with backup size more than 100 bytes:")
+    request = backup_pb.ListBackupsRequest(
+        parent=instance.name, filter="size_bytes > 100"
+    )
+    operations = spanner_client.database_admin_api.list_backups(request)
+    for backup in operations:
+        print(backup.name)
+
+    # List backups that were created after a timestamp that are also ready.
+    create_time = datetime.utcnow().replace(microsecond=0) - timedelta(days=1)
+    print(
+        'All backups created after "{}-{}-{}T{}:{}:{}Z" and are READY:'.format(
+            *create_time.timetuple()
+        )
+    )
+    request = backup_pb.ListBackupsRequest(
+        parent=instance.name,
+        filter='create_time >= "{}-{}-{}T{}:{}:{}Z" AND state:READY'.format(
+            *create_time.timetuple()
+        ),
+    )
+    operations = spanner_client.database_admin_api.list_backups(request)
+    for backup in operations:
+        print(backup.name)
+
+    print("All backups with pagination")
+    # If there are multiple pages, additional ``ListBackup``
+    # requests will be made as needed while iterating.
+    paged_backups = set()
+    request = backup_pb.ListBackupsRequest(parent=instance.name, page_size=2)
+    operations = spanner_client.database_admin_api.list_backups(request)
+    for backup in operations:
+        paged_backups.add(backup.name)
+    for backup in paged_backups:
+        print(backup)
+
+
+# [END spanner_list_backups]
+
+
+# [START spanner_delete_backup]
+def delete_backup(instance_id, backup_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+    backup = spanner_client.database_admin_api.get_backup(
+        backup_pb.GetBackupRequest(
+            name="{}/backups/{}".format(instance.name, backup_id)
+        )
+    )
+
+    # Wait for databases that reference this backup to finish optimizing.
+    while backup.referencing_databases:
+        time.sleep(30)
+        backup = spanner_client.database_admin_api.get_backup(
+            backup_pb.GetBackupRequest(
+                name="{}/backups/{}".format(instance.name, backup_id)
+            )
+        )
+
+    # Delete the backup.
+    spanner_client.database_admin_api.delete_backup(
+        backup_pb.DeleteBackupRequest(name=backup.name)
+    )
+
+    # Verify that the backup is deleted.
+    try:
+        backup = spanner_client.database_admin_api.get_backup(
+            backup_pb.GetBackupRequest(name=backup.name)
+        )
+    except NotFound:
+        print("Backup {} has been deleted.".format(backup.name))
+        return
+    assert True is False
```

Comment:
Why is this here?

Reply:
Success path should go into "except NotFound" exception we want to fail exception is not raised
Can you please suggest better way to do it?

Reply:
This looks wierd since it gets dispalyed on the public sample.
Can we instead assert `Backup {} has been deleted` this in the test? If this statement is there then all good, if not then it means delete backup failed.